### PR TITLE
Use generic modal for all simple modals

### PR DESCRIPTION
**Pull request by arjunu**
Run #p3

### CODE REVIEW

File: packages/web/app/components/dashboard/my-portfolio/PortfolioStats.tsx [hunk 1]
```diff
@@ -96,7 +96,7 @@ const PortfolioStatsLayout: React.FunctionComponent<
     <>
       {assets.slice(0, 3).map(eto => (
         <DataRowSeparated
-          key={eto.equityTokenName}
+          key={eto.etoId}
```
Comment (arjunu) [inline packages/web/app/components/dashboard/my-portfolio/PortfolioStats.tsx:99]: there are duplicate token names in io fixtures

File: packages/web/app/modules/generic-modal/actions.ts [hunk 2]
```diff
@@ -1,60 +1,25 @@
...
-      "exclamation",
-      actionLinkText,
-      onClickAction,
-    ),
+export const genericModalActions = {
+  show: <T extends EGenericModalType>({
+    type,
+    componentProps,
+  }: {
+    type: T;
```
Comment (desfero) [inline packages/web/app/modules/generic-modal/actions.ts:13]: let's extract type to a separate variable `TModalConfig` and also let's make a type system correct by removing optional from `componnetProps` and pass explicitly `undefined` where it's required

File: packages/web/app/modules/generic-modal/reducer.ts [hunk 3]
```diff
@@ -1,53 +1,33 @@
...
-//Add more custom icons here
-export type TIconType = keyof typeof genericModalIcons;
-
-export interface IGenericModal {
-  title: TMessage;
-  description?: TMessage;
-  icon?: TIconType;
-  actionLinkText?: TMessage;
-  onClickAction?: AppActionTypes;
+  type?: EGenericModalType;
```
Comment (desfero) [inline packages/web/app/modules/generic-modal/reducer.ts:9]: i would say we should not use `optional` in-state but rather `| undefined` to explicitly set undefined values in `initialState` so it's easier to understand

File: packages/web/app/modules/generic-modal/selectors.ts [hunk 4]
```diff
@@ -1,29 +1,13 @@
...
-  selectGenericModal,
-  (state: DeepReadonly<IGenericModalState>): React.ComponentType<any> | undefined =>
-    state.component,
-);
-
-export const selectGenericModalComponentProps = createSelector(
-  selectGenericModal,
-  (state: DeepReadonly<IGenericModalState>): object | undefined => state.componentProps,
-);
+export const selectGenericModal = (state: TAppGlobalState) => ({
```
Comment (desfero) [inline packages/web/app/modules/generic-modal/selectors.ts:9]: let's use `reselect` here

File: packages/web/app/components/modals/generic-modal/GenericModal.tsx [hunk 5]
```diff
@@ -1,109 +1,258 @@
...
+};
+
+const renderModalComponentWithProps = ({
+  type,
+  componentProps,
+  closeModal,
+  dispatch,
+}: Omit<TProps, "isOpen">) => {
+  let content: React.ReactNode | null = null;
+  let onClose: (() => void) | undefined = closeModal;
```
Comment (desfero) [inline packages/web/app/components/modals/generic-modal/GenericModal.tsx:93]: do we need onClose variable to just hold the reference to closeModal?

**Review by desfero — CHANGES_REQUESTED** [review]

File: packages/web/app/components/modals/Modal.tsx [hunk 6]
```diff
@@ -41,16 +43,21 @@ export const Modal: React.FunctionComponent<IModalComponentProps & CommonHtmlPro
...
-        {onClose && (
-          <ButtonClose
-            data-test-id="modal-close-button"
-            onClick={onClose}
-            iconProps={{ alt: <FormattedMessage id="common.close" /> }}
-          />
-        )}
-      </div>
+    <Panel
+      className={cn(styles.modal, { [styles.fixed]: scrollable }, { [styles.noFrame]: noFrame })}
```
[the remark this answers is not in the document]
Comment (arjunu) [reply packages/web/app/components/modals/Modal.tsx:47]: updated

File: packages/web/app/components/modals/generic-modal/GenericModal.tsx (hunk 5 above)
Comment (arjunu) [reply packages/web/app/components/modals/generic-modal/GenericModal.tsx:93]: removed

File: packages/web/app/components/modals/generic-modal/GenericModal.tsx [hunk 7]
```diff
@@ -1,109 +1,258 @@
...
+const GenericModalLayout: React.FunctionComponent<TProps> = props => {
+  if (props.isOpen) {
+    invariant(props.type, "type is required for generic modal");
+
+    return props.componentProps
+      ? renderModalComponentWithProps(props)
+      : renderModalComponentWithoutProps(props);
+  }
+
+  return <Modal isOpen={false} />;
```
[the remark this answers is not in the document]
Comment (arjunu) [reply packages/web/app/components/modals/generic-modal/GenericModal.tsx:243]: it's for close animation, if we return null then it is removed from DOM immediately

File: packages/web/app/components/modals/unverified-email-reminder/UnverifiedEmailReminderModal.tsx [hunk 8]
```diff
@@ -1,54 +1,57 @@
...
+    case EWalletType.LEDGER:
       return (
         <FormattedMessage
           id="checkUnverifiedEmailModal.browser-ledger"
           values={{ unverifiedEmail }}
         />
       );
     default:
-      assertNever(modalType);
+      assertNever(
```
[the remark this answers is not in the document]
Comment (arjunu) [reply packages/web/app/components/modals/unverified-email-reminder/UnverifiedEmailReminderModal.tsx:34]: yes better, updated

File: packages/web/app/modules/generic-modal/selectors.ts (hunk 4 above)
Comment (arjunu) [reply packages/web/app/modules/generic-modal/selectors.ts:9]: updated

File: packages/web/app/modules/generic-modal/reducer.ts (hunk 3 above)
Comment (arjunu) [reply packages/web/app/modules/generic-modal/reducer.ts:9]: updated

File: packages/web/app/modules/auth/email/sagas.ts [hunk 9]
```diff
@@ -145,19 +148,29 @@ export function* checkForPendingEmailVerification(): Generator<any, void, any> {
...
+        componentProps: {
+          unverifiedEmail,
+          walletType,
+        },
       }),
     );
   }
 }
 
+export function* checkTOSAgreement(): SagaGenerator<void> {
```
[the remark this answers is not in the document]
Comment (arjunu) [reply packages/web/app/modules/auth/email/sagas.ts:167]: nice catch, moved

File: packages/web/app/components/modals/verify-bank-account-cta-modal/VerifyBankAccountCtaModal.tsx [hunk 10]
```diff
@@ -14,6 +14,7 @@ type TVerifyBankAccountCtaModalProps = {
   verifyUserModalIsOpen: boolean;
 };
 
+// TODO use generic modal once it migrates to new design
```
[the remark this answers is not in the document]
Comment (arjunu) [reply packages/web/app/components/modals/verify-bank-account-cta-modal/VerifyBankAccountCtaModal.tsx:17]: https://github.com/Neufund/platform-frontend/issues/4960

File: packages/web/app/modules/generic-modal/actions.ts (hunk 2 above)
Comment (arjunu) [reply packages/web/app/modules/generic-modal/actions.ts]: done

File: packages/web/app/components/modals/generic-modal/GenericModal.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (desfero) [reply packages/web/app/components/modals/generic-modal/GenericModal.tsx:242]: ahhh, i see

File: packages/web/app/components/modals/verify-bank-account-cta-modal/VerifyBankAccountCtaModal.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (desfero) [reply packages/web/app/components/modals/verify-bank-account-cta-modal/VerifyBankAccountCtaModal.tsx:17]: <3

**Review by desfero — APPROVED** [review]

**Pull request merged**